### PR TITLE
[DR-111] Quote Copy-to-clipboard indicator

### PR DESCRIPTION
**Pull request by blurbyte**
# [DR-111](https://digitalrig.atlassian.net/browse/DR-111) Quote Copy-to-clipboard indicator

- [x] Added notification after quote is copied
- [x] Updated unit and e2e tests

copy-notification

**TODO (kindof)**
Figure out how to make 1 commented out e2e test work

### CODE REVIEW

File: cypress/integration/QuoteBubble.test.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  })
+
+  describe('when Twitter share link is clicked', () => {
+    it('should open suitable Twitter share window', () => {
+      cy.getByTestId('twitter-share-button').click();
+      cy.get('@windowOpen').should('be.calledWithMatch', 'http://www.twitter.com/');
+    });
+  })
+
+  // describe('when Copy-to-clipboard button is clicked ', () => {
```
Comment (michaelmoney) [inline cypress/integration/QuoteBubble.test.js]: Did you leave this comment for a purpose? 🤔

Comment (blurbyte) [reply cypress/integration/QuoteBubble.test.js]: Yeah it's about TODO, looking for volunteers to pick it up :D

File: src/components/QuoteBubble/CopyNotification.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+const Wrapper = styled(animated.div)`
+  position: absolute;
+  top: 1.4rem;
+  left: 1.4rem;
+  transform-origin: 50% 100%;
+  z-index: ${zIndexArtworkPart};
+`;
+
+function CopyNotification({ isVisible, onFinished }) {
+  const transitions = useTransition(isVisible, null, {
```
Comment (mgolawski) [inline src/components/QuoteBubble/CopyNotification.js]: I think that you can do that animation way easier with `useSpring`, and similar logic as we did in `Button.js`, or `ButtonBase.js`

File: src/components/QuoteBubble/__tests__/BubbleButtons.test.js [hunk 3]
```diff
@@ -1,17 +1,29 @@
...
-  it('renders correctly for Facebook variant', () => {
+  it('renders correctly if navigator share API is availible', () => {
+    global.navigator.share = jest.fn();
+
     const component = create(<BubbleButtons quote={'Test Quote'} />);
 
     expect(component.toJSON()).toMatchSnapshot();
   });
+
+  it('renders correctly if navigator share API is NOT availible', () => {
```
Comment (mgolawski) [inline src/components/QuoteBubble/__tests__/BubbleButtons.test.js]: :+1: forgot about that case.

File: src/components/QuoteBubble/__tests__/BubbleButtons.test.js [hunk 4]
```diff
@@ -1,17 +1,29 @@
...
-
-jest.mock();
+jest.mock('../CopyButton', () => 'CopyButton');
+jest.mock('../DesktopShare', () => 'DesktopShare');
+jest.mock('../MobileShare', () => 'MobileShare');
 
 describe('COMPONENT - QuoteBubble BubbleButtons', () => {
-  it('renders correctly for Facebook variant', () => {
+  it('renders correctly if navigator share API is availible', () => {
+    global.navigator.share = jest.fn();
```
Comment (mgolawski) [inline src/components/QuoteBubble/__tests__/BubbleButtons.test.js]: Either we declare `navigator` here, or in `setupTests.js`. If you want to keep it here, please remove the declaration from setup file.

File: src/components/QuoteBubble/CopyNotification.js (hunk 2 above)
Comment (blurbyte) [reply src/components/QuoteBubble/CopyNotification.js]: According to documentation for **mount/unmount single-component reveals** should be handled by **useTransition**.

File: src/components/QuoteBubble/__tests__/BubbleButtons.test.js (hunk 4 above)
Comment (blurbyte) [reply src/components/QuoteBubble/__tests__/BubbleButtons.test.js]: It's just here for clarity.

**Pull request merged**